packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -763,12 +763,15 @@ dynamic _toEncodable(dynamic object) {
 }
 
 Future<Map<String, dynamic>> _deviceToMap(Device device) async {
+  final bool isLocalEmulator = await device.isLocalEmulator;
   return <String, dynamic>{
     'id': device.id,
     'name': device.name,
     'platform': getNameForTargetPlatform(await device.targetPlatform),
-    'emulator': await device.isLocalEmulator,
+    'emulator': isLocalEmulator,
     'category': device.category.toString(),
+    'platformType': device.platformType.toString(),
+    'ephemeral': device.ephemeral ?? isLocalEmulator,
```

Comment:
Does this result in Android mobiles (non-emulators) being non-ephemeral? That would mean we wouldn't auto-select them when you plug one in?